packages/sources/finnhub/src/endpoint/quote.ts

```diff
@@ -79,39 +78,52 @@ export const buildSymbol = ({
-  req.requestContext.data.quote = req.requestContext.data.quote?.toUpperCase()
-  req.requestContext.data.exchange = req.requestContext.data.exchange?.toUpperCase()
+const requestTransform =
+  (defaultExchange: string) => (req: AdapterRequest<typeof inputParameters.validated>) => {
+    req.requestContext.data.base = req.requestContext.data.base.toUpperCase()
+    req.requestContext.data.quote = req.requestContext.data.quote?.toUpperCase()
+    req.requestContext.data.exchange = req.requestContext.data.exchange?.toUpperCase()
 
-  const requestBase = req.requestContext.data.base
-  const requestExchange = req.requestContext.data.exchange
+    const requestBase = req.requestContext.data.base
+    const requestQuote = req.requestContext.data.quote
+    const requestExchange = req.requestContext.data.exchange
 
-  // If the symbol is a full symbol (containing Exchange and Pair), split into the separate params
-  if (isExchangeSymbol(requestBase)) {
-    const params = splitSymbol(requestBase)
+    // If the symbol is a full symbol (containing Exchange and Pair), split into the separate params
+    if (isExchangeSymbol(requestBase)) {
+      const params = splitSymbol(requestBase)
 
-    // If exchange is provided in request, prefer that over the exchange split from the symbol
-    const exchange = requestExchange || params.exchange
+      // If exchange is provided in request, prefer that over the exchange split from the symbol
+      const exchange = requestExchange || params.exchange
 
-    if (exchange && exchanges[exchange] == undefined) {
-      throw new AdapterInputError({
-        statusCode: 400,
-        message: `${exchange} is not a supported exchange, must be one of [${Object.keys(
-          exchanges,
-        ).join(', ')}]`,
-      })
+      req.requestContext.data.base = params.base
+      req.requestContext.data.quote = params.quote
+      req.requestContext.data.exchange = exchange
+
+      return
     }
 
-    req.requestContext.data.base = params.base
-    req.requestContext.data.quote = params.quote
-    req.requestContext.data.exchange = exchange
+    // Use default exchange if none is provided
+    if (requestBase && requestQuote) {
+      req.requestContext.data.exchange = requestExchange || defaultExchange
+      return
+    }
+  }
 
-    return
+// Check that the exchange is in our list of exchange configs, else throw an error
+const validateExchange = (req: AdapterRequest<typeof inputParameters.validated>) => {
+  const exchange = req.requestContext.data.exchange
+
+  if (exchange && exchanges[exchange] == undefined) {
+    throw new AdapterInputError({
+      statusCode: 400,
+      message: `'${exchange}' is not a supported exchange, must be one of [${Object.keys(
+        exchanges,
+      ).join(', ')}]`,
+    })
   }
 }
 
-export const buildQuoteEndpoint = (overrides?: Record<string, string>) =>
+export const buildQuoteEndpoint = (defaultExchange: string) =>
```

Comment:
This is slightly misleading terminology - oanda & finnhub aren't exchanges, I'd probably use something like 'source' or 'provider' instead

Reply:
This comes from Finnhub's docs, which refer to these as exchanges: https://finnhub.io/docs/api/forex-exchanges

Happy to update to `source`, but will need to sync with Data Products as that will involve also updating the [exchange input parameter](https://github.com/smartcontractkit/external-adapters-js/blob/main/packages/sources/finnhub/src/endpoint/quote.ts#L27). 

Are you happy to continue using `exchange` for now, and I'll put up a follow-up PR once I've discussed with Data Products?